Intento de modelo predictivo

In [4]:
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.ensemble import RandomForestRegressor
from sklearn.metrics import mean_squared_error, r2_score

# Cargar los datos
data = pd.read_csv('FixData/6to/6TO_COMU.csv', delimiter=';')

# Convertir variables categóricas en variables dummy
data = pd.get_dummies(data)

X = data.drop(['PROMEDIO_X_SECCION'], axis=1)
y = data['PROMEDIO_X_SECCION']

# Conjuntos de entrenamiento y prueba
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

# Inicializar y entrenar el modelo
model = RandomForestRegressor()
model.fit(X_train, y_train)

# Realizar predicciones en el conjunto de prueba
y_pred = model.predict(X_test)

# Evaluar el modelo
mse = mean_squared_error(y_test, y_pred)
r2 = r2_score(y_test, y_pred)

print(f'Mean Squared Error: {mse}')
print(f'R^2 Score: {r2}')


Mean Squared Error: 29.334907692307695
R^2 Score: 0.5070988864585404


Mean Squared Error (MSE):

El MSE es una medida de la calidad del ajuste de un modelo de regresión a los datos observados.
Cuanto menor sea el valor del MSE, mejor será el rendimiento del modelo.
MSE se calcula como la media de los cuadrados de las diferencias entre los valores predichos y los valores reales.
Para este ejercicio, el MSE es aproximadamente 29.33. Esto significa que, en promedio, el cuadrado de la diferencia entre las predicciones del modelo y los valores reales es de alrededor de 29.33. Un MSE más bajo indica que el modelo está haciendo predicciones más precisas.

R^2 Score:

El R^2 Score, o coeficiente de determinación, es una medida que indica qué tan bien las predicciones del modelo se ajustan a los datos reales.
El valor de R^2 está entre 0 y 1. Un valor más cercano a 1 indica un mejor ajuste del modelo a los datos, mientras que un valor más cercano a 0 indica que el modelo no se ajusta bien a los datos.
R^2 = 1 significa que el modelo explica perfectamente la variabilidad de los datos.
R^2 = 0 significa que el modelo no explica ninguna variabilidad de los datos.
En este caso, el R^2 Score es aproximadamente 0.51. Esto significa que el modelo explica alrededor del 51% de la variabilidad de los datos, lo cual indica un ajuste moderado del modelo a los datos.